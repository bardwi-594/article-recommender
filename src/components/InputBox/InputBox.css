form.input-form input {
    line-height: 1.2em;
    padding: 13px 24px;
    font-size: 0.875rem;
    background: #fff;
    border: 1px solid rgba(34,36,38,.15);
    color: rgba(0,0,0,.87);
    border-radius: .3rem;
    box-shadow: 0 0 0 0 transparent inset;
    transition: color .1s ease,border-color .1s ease;
}

form.input-form input:focus {
    color: rgba(0,0,0,.95);
    border-color: #1d96e8;
    border-radius: .3rem;
    outline: none;
    background: #fff;
    box-shadow: 0 0 0 0 rgb(34 36 38 / 35%) inset;
}

div.inputbox-container{
    display: flex;
    justify-content: center;
}

div.form-field {
    display: flex;
    flex-direction: row;
    margin: 0 -.5em 1em;
}

div.form-field input {
    margin-right: 0.5em;
}

div button.submit {
    background-color: #1d96e8;
    border: 0;
    display: inline-block;
    color: #fff;
    cursor: pointer;
    font-size: 0.9375rem;
    font-weight: 600;
    letter-spacing: .03em;
    padding: 13px 24px;
    border-radius: 0.3rem;
    transition: .1s ease;
}

div button.submit:focus {
    outline: none;
    opacity: 1;
}