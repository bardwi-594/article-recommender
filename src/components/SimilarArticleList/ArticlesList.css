/**CSS code for the Article List Component**/

div.list-container {
    max-width: 900px;
    margin: 0px auto;
}

div.list-container:last-child {
    padding-bottom: 0;
}

div.heading {
    margin: 1rem auto;
    max-width: 700px;
}


/**Code for screen size min-width: 280px**/
@media only screen and (min-width: 280px){
    div.heading {
        margin: 30px;
        max-width: 700px;
    }
    div.list-container {
        max-width: 900px;
        margin: 0px auto;
    }
    
    div.list-container:last-child {
        padding-bottom: 0;
    }
}

/**Code for screen size min-width: 500px**/
@media only screen and (min-width: 500px){
    div.heading {
        margin: 30px;
        max-width: 700px;
    }
    div.list-container {
        max-width: 900px;
        margin: 0px auto;
    }
    
    div.list-container:last-child {
        padding-bottom: 0;
    }
 
}

/**Code for screen size min-width: 600px**/
@media only screen and (min-width: 600px){
    div.heading {
        margin: 30px;
        max-width: 700px;
    }
    div.list-container {
        max-width: 900px;
        margin: 0px auto;
    }
    
    div.list-container:last-child {
        padding-bottom: 0;
    }
}

/**Code for screen size min-width: 768px**/
@media only screen and (min-width: 768px){
    div.heading {
        margin: 1rem auto;
        max-width: 700px;
    }
    div.list-container {
        max-width: 900px;
        margin: 0px auto;
    }
    
    div.list-container:last-child {
        padding-bottom: 0;
    }
}