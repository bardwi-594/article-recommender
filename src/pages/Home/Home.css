/**CSS Code for Component for Home Screen**/

.header {
    text-align: center;
    margin: 20px;
}

p.loader-text {
    text-align: center;
}

.input-search {
    height: 100px;
}

/**Style for the loading element**/
p.loader {
   border: 2px solid #f3f3f3;
   border-top: 2px solid #b9c4cc;
   position: relative;
   left: 50%;
   width: 20px;
   height: 20px;
   border-radius: 50%;
   animation: spin 2s linear infinite;
}

@keyframes spin {
    0% { transform: rotate(0deg); }
    100% { transform: rotate(360deg); }
  }
  
  p.error-message {
      text-align: center;
  }
/**Style for the loading element**/







