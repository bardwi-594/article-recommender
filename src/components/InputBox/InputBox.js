import React, { Component } from "react";
import "./InputBox.css";

class InputBox extends Component {
    render() {
        return (
            <div className="inputbox-container">
                <form className="input-form">
                    <div className="form-field">
                        <input />
                        <button className="submit" type="submit" color="green">Submit</button>
                    </div>
                </form>

            </div>
        )
    }
}
export default InputBox;