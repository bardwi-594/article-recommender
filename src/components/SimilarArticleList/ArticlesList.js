import React from "react";
import SimilarArticles from "../SimilarArticles/SimilarArticles";
import Filters from "../Filters/Filters";
import "./ArticlesList.css";

/***Component for the rendering the Lis of similar Articles ***/
const ArticlesList = (props) => {
    return (
        <div className="list-container">
            <div className="heading">
                <h1>Similar articles</h1>
                <Filters />  {/**Component for the Filters **/}
            </div>
            {/** With props.similararticles passing down the array of the similar articles as a prop**/}
            {/** Using map() method on props.similarearticles we get a list of similar articles **/}

            {props.similararticles.map((article, document_id) => (
                <SimilarArticles
                    article={article}
                    title={article.title}
                    document_id={article.document_id}
                    key={article.title + document_id} />
            ))}
            {/** Passing the key props to SimilarArticles component **/}
        </div>
    )
}

export default ArticlesList;