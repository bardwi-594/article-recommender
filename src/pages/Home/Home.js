import React, { Component } from 'react';
import InputBox from "../../components/InputBox/InputBox";
import getsimilararticle from "../../api/api";
import ArticlesList from "../../components/SimilarArticleList/ArticlesList";
import "./Home.css";

/** Code for the Home screen **/
class Home extends Component {
    state = {
        similararticles: [], /** to store an array of articles**/
        error: "", /** variable to handle any errors we receive from the API**/
        input: "", /** variable to hold the urlInput**/
        loading: false, /** Setting the initial value for loader to false by default **/
    }

    /** inputTextBox is the method to handle the state **/
    inputTextBox = async (urlInput) => {
        try {
            this.setState({
                loading: true /** Loading is set to true when the method is called **/
            });
            const response = await getsimilararticle(urlInput);
            /** When API executes successfully the similararticles is updated in the state from the response**/
            this.setState({
                similararticles: response,
                input: urlInput,
            });
        } catch (error) {
            this.setState({ error: "Sorry! Could not find any similar articles" });
            /** The error state will be updated with the message when there is no result or response **/
        }
        this.setState({
            loading: false
        });
        /** Once the method is executed the loading state will update to false **/
    }

    render() {
        const { similararticles, error, loading } = this.state;

        return (
            <div className="container">
                <header className="header">
                    Search text
                </header>
                <div className="input-search">
                    {/**InputBox-Component for the Input text box **/}
                    <InputBox
                        inputTextBox={this.inputTextBox}
                    />
                    {/**When the loading state is true it will render the loader**/}
                    {loading && (
                        <div className="loading">
                            <p className="loader-text">Searching for similar articles</p>
                            <p className="loader"></p>

                        </div>
                    )}
                </div>
                {/*** If similar articles exists then it will render ArticlesList Component with render***/}
                {similararticles.length > 0 &&
                    <ArticlesList similararticles={similararticles} />}
                {/*** If there is any error, it will render an error message***/}
                {error &&
                    <p className="error-message">Sorry, could not get any results! Please try again.</p>
                }

            </div>
        )
    }

}

export default Home;