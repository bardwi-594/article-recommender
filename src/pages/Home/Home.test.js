import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, render, mount, configure } from 'enzyme';
import Home from "./Home";
import InputBox from "../../components/InputBox/InputBox";
import ArticlesList from "../../components/SimilarArticleList/ArticlesList";

configure({ adapter: new Adapter() });

const mockdata = [
    {
        url: "www.google.com",
        title: "travel"
    }
];

describe("Home Component", () => {
    it("renders", () => {
        const wrapper = shallow(<InputBox />);
        expect(wrapper.exists()).toBe(true);
    });

    it("Input text is echoed", () => {
        const wrapper = shallow(<InputBox inputTextBox={() => { }} />);
        wrapper.find("input").simulate("change", {
            target: {
                value: "url"
            }
        });
        expect(wrapper.find("input").props().value).toEqual("url");
    });

    it("Similar articles list", () => {
        const wrapper = shallow(<ArticlesList similararticles={mockdata} />);
        expect(wrapper).toMatchSnapshot();

    })



})


