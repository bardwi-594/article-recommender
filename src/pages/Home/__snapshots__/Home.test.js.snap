// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component Similar articles list 1`] = `ShallowWrapper {}`;
